.azure-pipelines/azure-pipelines-image-template.yml

```diff
@@ -38,6 +38,7 @@ jobs:
         submodules: recursive
         displayName: 'Checkout code'
       - script: |
+          git checkout -b $(Build.SourceBranchName)
```

Comment:
i have some concern on this one. the checkout behavior is complicated when we deal will pr build. we should only fetch the branch, but should not modify the checkout behavior. this won't work for pr build.

Reply:
There is a -b option, means to create a new branch. It should not change the current commit.
git checkout -b <branchname>
Different from: 
git checkout <branchname>

Reply:
The branch name is merge, and the commit not change see detail in draft PR: https://github.com/Azure/sonic-buildimage/pull/7329 and the build result: https://dev.azure.com/mssonic/build/_build/results?buildId=10941&view=logs&j=83516c17-6666-5250-abde-63983ce72a49&t=1f6d5e21-f0ce-508d-13df-d1c50f7ea7b9

* (HEAD detached at pull/7329/merge) 482f951a Merge c2486425cac936b5ce1834a433922226fc570377 into 15487f4b5e48d9d5dcf3433997be30beb8d33c03
SourceBranchName=merge
Switched to a new branch 'merge'
* merge 482f951a Merge c2486425cac936b5ce1834a433922226fc570377 into 15487f4b5e48d9d5dcf3433997be30beb8d33c03

Reply:
the branch name "merge" is not correct one.